File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
  if (!root) return []
  let level = [root]
  const res = []
  
  while (level.length) {
    res.push(level.map((node) => node.val))
    level = level.flatMap((node) => node.children)
  }
  
  return res
};

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func levelOrder(root *Node) [][]int {
    var res [][]int
    
    if root == nil {
        return res
    }
    
    level := []*Node{ root }
    
    for len(level) > 0 {
        size := len(level)
        var vals []int
        
        for _, node := range level {
            vals = append(vals, node.Val)
            level = append(level, node.Children...)
        }
        
        res = append(res, vals)
        level = level[size:]
    }
    
    return res
}